tests/fit_data/test_save_data_to_file.py

```diff
@@ -0,0 +1,77 @@
+from pathlib import Path
```

Comment:
Those are actually very solid **integration tests**.

One key element of **unit tests** is that they don't interact with the operating system (aka, read and write files).
*Integration tests* however, do exactly that. They test the software end-to-end and see that the different components combine nicely together.

Please keep those tests as they are, but move them to the *integration_tests* directory (if it's missing in your branch, update the master branch from upstream, and merge it into your feature branch. I added this directory yesterday).

After you do that, write new unit tests, but use [mocks](https://realpython.com/python-mock-library/) in order to simulate writing to a file.

Mocks can sometimes be perceived as a little bit hard concept for developers who are not familiar with them. Please feel free to say that you rather I will write the unit tests for you, and I'll step right in.